makefiles/docker.inc.mk

```
@@ -89,11 +89,6 @@ DOCKER ?= docker
 DOCKER_VOLUMES_AND_ENV += $(if $(wildcard $(GIT_CACHE_DIR)),-v $(GIT_CACHE_DIR):$(DOCKER_BUILD_ROOT)/gitcache)
 DOCKER_VOLUMES_AND_ENV += $(if $(wildcard $(GIT_CACHE_DIR)),-e GIT_CACHE_DIR=$(DOCKER_BUILD_ROOT)/gitcache)
 
-# Handle worktree by mounting the git common dir in the same location
-_is_git_worktree = $(shell grep '^gitdir: ' $(RIOTBASE)/.git 2>/dev/null)
-GIT_WORKTREE_COMMONDIR = $(shell git rev-parse --git-common-dir)
```

Instead of removing these lines, just convert it to absolute path and it should work again:

```suggestion
GIT_WORKTREE_COMMONDIR = $(abspath $(shell git rev-parse --git-common-dir))
```

---

the latter would also work with older git versions, where neither `--git-common-dir` nor `--absolute-git-dir` are present

---

`--absolute-git-dir` does not give us what we want. In the example below I'm on a worktree (masterRIOT). The "real" repo is vanillaRIOT.
```
~/source/masterRIOT/examples $ git rev-parse --absolute-git-dir
/home/jcarrano/source/vanillaRIOT/.git/worktrees/masterRIOT
```
Same for `--git-dir`.

---

Because the `.git/worktrees/work_tree_name` was not enough I think. I will test to verify

---

With `+GIT_WORKTREE_COMMONDIR = $(shell git rev-parse --absolute-git-dir)` I get an `UNKNOWN` version and the same with `--git-dir`.

---

Hmm I should refresh the page before answering ><

---

okay, I'll will try that `abspath` thingy and will adapt this accordingly, thx for the feedback and explanation why this was needed.